src/engine/Core.java

```diff
@@ -157,17 +160,24 @@ public static void main(final String[] args) {
 						+ gameState.getBulletsShot() + " bullets shot and "
 						+ gameState.getShipsDestroyed() + " ships destroyed.");
 				currentScreen = new ScoreScreen(width, height, FPS, gameState);
+
+				/** translate 10% of score to coin */
+				wallet.deposit(gameState.getScore()/10);
```

Comment:
It would be better to utilize CoinEarned rather than fetching the score directly from the gameState. Currently, CoinEarned is empty, which causes the ScoreScreen to show 0 coins earned, and it would be more useful to use the CoinEarned field to modify the earned percentage in the future or to increase the earned percentage based on the level of coins earned.

gameState 에서 Score를 직접 불러와서 획득하기 보다는 보다는 CoinEarned를 활용하는게 더 좋을것 같습니다. 현재 CoinEarned가 비어있어 ScoreScreen 에서도 획득한 Coin이 0으로 표시되고 있으며, 추후 획득 비율을 수정하거나 Coin 획득량 레벨에 따라 획득 비율을 올릴때에도 CoinEarned 필드를 사용 하는게 더 유용해 보입니다.

Reply:
I removed the coinEarned part from gameState since coins are not processed during gameplay. I’ve modified it so that coins can be handled through CoinEarned in ScoreScreen!

게임 플레이 도중 코인을 처리하는 것은 아니기에 gameState의 coinEarned부분은 제거했습니다. ScoreScreen에서 CoinEarned를 통해 코인 획득을 처리할 수 있도록 수정했습니다!